services/src/main/java/org/keycloak/protocol/oidc/mappers/AbstractPairwiseSubMapper.java

```diff
@@ -90,7 +90,7 @@ protected void setAccessTokenSubject(IDToken token, String pairwiseSub) {
     }
 
     protected void setUserInfoTokenSubject(IDToken token, String pairwiseSub) {
-        token.setSubject(pairwiseSub);
+        token.getOtherClaims().put("sub", pairwiseSub);
```

Comment:
Not sure I understand the reason behind this change

Reply:
Only reverted the change I did in the previous PR.

Reply:
Seems like the change you made here was valid, but not sure tbh, as this stuff really is a bit odd.

Reply:
Yes, but I would prefer reverting it as it was before. It was not really related to the original issue.

Reply:
I agree that this is strange workaround, however it seems as the safest what can be done IMO.

The problem is one bigger issue that UserInfoTokenMapper class accepts "AccessToken" as an argument for some reason. This doesn't makes sense to me as it should instead accept "UserInfo" representation as an argument. This will allow to cleanup these workarounds and the strange code in UserInfoEndpoint, which need to manually convert token into the "claims" map. I've created JIRA for this few days ago https://issues.redhat.com/browse/KEYCLOAK-15537 . I suggest to address that one as a follow-up to remove workarounds. For now, I am just accepting the PR as it seems as the safest way for the backport to z-stream.